options.go

```diff
@@ -164,12 +164,36 @@ func (o *Option[T]) Scan(src any) error {
 	if err != nil {
 		return fmt.Errorf("Option[%T].Scan: failed to convert value from SQL driver: %w", o.value, err)
 	}
-	v, ok := av.(T)
-	if !ok {
-		return fmt.Errorf("Option[%T].Scan: failed to convert value '%#v' to type %T", o.value, v, o.value)
+	return convertType(av, o)
+}
+
+func convertType[T any](src any, dst *Option[T]) error {
+	// First, try to directly convert the value to T
+	v, ok := src.(T)
+	if ok {
+		*dst = New(v)
+		return nil
 	}
-	*o = New(v)
-	return nil
+
+	// Convert between string and []byte
+	dstType := reflect.TypeOf(dst.value)
+	srcType := reflect.TypeOf(src)
+	if isByteSlice(srcType) && dstType.Kind() == reflect.String {
+		src = string(src.([]byte))
+		*dst = New(src.(T))
+		return nil
+	}
+	if srcType.Kind() == reflect.String && isByteSlice(dstType) {
+		src = []byte(src.(string))
+		*dst = New(src.(T))
+		return nil
+	}
+
+	return fmt.Errorf("Option[%T].Scan: failed to convert value %#v to type %T", dst.value, src, dst.value)
```

Comment:
It seems that the dst and src type mismatch problem is not limited to strings.
https://zenn.dev/methane/articles/2023-go-mysql-parse-numbers

If we wanted full SQL Driver support with Option type, the flow would be something like this

1. if `src` is `nil`, `dst` becomes `None`
2. if `dst` type implements a Scanner interface, use it
3. if `dst` type is a primitive (or its defined type), use a huge switch statement to achieve the conversion, just as the sql package does. https://cs.opensource.google/go/go/+/master:src/database/sql/convert.go;l=211?q=convertAssign&ss=go%2Fgo

Or, we may be able to use `sql.Null[T]` after releasing go 1.22.
https://github.com/golang/go/issues/60370